libguard/guard_interface.cpp

```diff
@@ -167,9 +171,20 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
             }
             return getHostEndiannessRecord(existGuard);
         }
+
+        // Storing the 1st resolved guard record position
+        if (existGuard.recordId == GUARD_RESOLVED)
+        {
+            if (!isResolvedRecordPresent)
+            {
+                empPos = pos;
+                isResolvedRecordPresent = true;
```

Comment:
Do we really need this variable? 
Just we cannot use `empPos`?
- If can, can you change the name of the variable for `empPos`?

Reply:
Suppose there are multiple invalid records so every time the empPos will get changed, we want 1st invalid record location/position. Any better idea please feel free to share.

Reply:
I think you can initialize the `empPos` with `-1` (we can not use `0` because there may be a chance that the first record can be resolved - I guess If you have a header then `0` we won't get it) and check with `-1` before assigning the first resolved position so that we can avoid that `isResolvedRecordPresent` variable i.e we can use `empPos == -1` instead of `! isResolvedRecordPresent`.

Reply:
Done.